app/components/ActivityBar.js

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable react/prefer-stateless-function */
+import React, { Component } from 'react';
+import {
+  StyleSheet, View,
+} from 'react-native';
+
+export default class ActivityBar extends Component {
+  render() {
+    let { activityLevel, barWidth } = this.props;
+
+    barWidth = parseInt(barWidth);
+
+    if (activityLevel === 'N/A') {
+      return (
+        <View style={[styles.outside, { width: barWidth }, { backgroundColor: 'gray' }]} />
+      );
+    }
+    else {
+      // Cap Activity Level at 100% 
+      if (parseInt(activityLevel) >= 100) {
+        activityLevel = '100';
+      }
```

Comment:
do all of this activity level checking before, and then use the previously mentioned short hand if else to return the view.